### PR TITLE
Opal 603 - Instruction to run argo tests locally

### PR DESCRIPTION
**Pull request by AM-USAF**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
+    - `kubectl apply -n pivot task-queue/resources/test_workflow_template.yaml`
+    - If you have the Argo CLI installed, you can use:
+    - `argo -n pivot template create resources/test_workflow_template.yaml`
+3. Port-forward the argo workflows server pod
+    - `shift+f` on K9s or `kubectl port-forward -n pivot service/argo-server 2746:2746`
+5. pip install the `task-queue` package (you will likely need to run this in a python virtual environment)
+6. `python -m pytest task-queue/tests/test_argo_workflows_worker_interface.py`
 
 
 ### S3 Connection
```
Comment (JR-USAF) [inline README.md:96]: Why was this removed?

File: README.md [hunk 2]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
+6. `python -m pytest task-queue`
+
+### Running Argo Workflow Tests in minikube
+
+1. Spin up the pods:
+    - `minikube start`
+    - `kubectl create namespace pivot`
+    - `kubectl apply -n pivot -f task-queue/resources/quick-start-minimal.yaml`
+2. Make sure the test workflow template is deployed to argo
+    - `kubectl apply -n pivot task-queue/resources/test_workflow_template.yaml`
```
Comment (JR-USAF) [inline README.md:87]: I think you forgot the `-f` flag on this. 

`kubectl apply -n pivot -f task-queue/resources/test_workflow_template.yaml`

File: README.md [hunk 3]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
-5. pip install the `data_pipeline` package
-6. `python -m pytest data_pipeline`
+    - `docker compose -f task-queue/resources/docker-compose.test.yaml up`
+5. pip install the `task-queue` package
+6. `python -m pytest task-queue`
+
+### Running Argo Workflow Tests in minikube
+
+1. Spin up the pods:
+    - `minikube start`
```
Comment (JR-USAF) [inline README.md:83]: Maybe add a bullet saying download minikube and give the download link.
https://minikube.sigs.k8s.io/docs/start/?arch=%2Fwindows%2Fx86-64%2Fstable%2F.exe+download

File: README.md [hunk 4]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
+    - `kubectl create namespace pivot`
```
Comment (JR-USAF) [inline README.md:84]: For me on windows I had to run `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V-All` and then `minikube start --vm-driver hyperv` to get minikube started. Just running `minikube start` didn't work. Not sure if it is worth putting platform specific stuff in this README though?

File: README.md [hunk 5]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
+An external MinIO or S3 service is required for the tests to run and `AWS_ACCESS_KEY_ID`
+`AWS_SECRET_ACCESS_KEY` and `FSSPEC_S3_ENDPOINT_URL` environment variables must be set.
 
 # Starting Work Queue Server and CLI
 
 You will need docker installed on you machine to use the Task Queue.
 
 ## Downloading Docker Image
 
-To download the docker image, go to the registry and select the desired image tag (https://github.com/AFMC-MAJCOM/task-queue/pkgs/container/task-queue%2Ftask-queue). The image tags mirror the version associated with the task-queue python package installed inside the image. For this example, the `latest` image will be used.
```
[the remark this answers is not in the document]
Comment (AM-USAF) [reply README.md:98]: I don't know! It was an accident, I think my working file overwrote a change in main, I'll fix this

File: README.md (hunk 3 above)
Comment (AM-USAF) [reply README.md:83]: I don't think I should include that in this readme, the expectation for running tests on AWS is that they already have kubectl and docker installed, I think adding instructions for installing minikube would mean we should add instructions for installing everything.

File: README.md (hunk 4 above)
Comment (AM-USAF) [reply README.md:84]: That's weird, I was able to use minikube start with windows, were you using WSL?

File: README.md [hunk 6]
```diff
@@ -55,47 +55,55 @@ Jobs are created as workflows via the submit API endpoint, and are monitored fro
...
+    - `kubectl create namespace pivot`
+    - `kubectl apply -n pivot -f task-queue/resources/quick-start-minimal.yaml`
+2. Make sure the test workflow template is deployed to argo
+    - `kubectl apply -n pivot task-queue/resources/test_workflow_template.yaml`
+    - If you have the Argo CLI installed, you can use:
+    - `argo -n pivot template create resources/test_workflow_template.yaml`
+3. Port-forward the argo workflows server pod
+    - `shift+f` on K9s or `kubectl port-forward -n pivot service/argo-server 2746:2746`
+5. pip install the `task-queue` package (you will likely need to run this in a python virtual environment)
+6. `python -m pytest task-queue/tests/test_argo_workflows_worker_interface.py`
```
[the remark this answers is not in the document]
Comment (AM-USAF) [reply README.md:93]: Yep, true, I forgot to add that, thank you!

File: README.md (hunk 1 above)
Comment (AM-USAF) [reply README.md:96]: Correct, fixed it

File: README.md (hunk 4 above)
Comment (AM-USAF) [reply README.md:84]: Ok, I'll generalize it

**Review by JR-USAF — APPROVED** [review]

**Review by JR-USAF — DISMISSED** [review]

**Review by JR-USAF — APPROVED** [review]

**Review by BE-USAF — APPROVED** [review]
